File: apps/ios/BitOS/DesignSystem/AppIcons.swift
import SwiftUI

/**
 * AppIcons — the application's single source of truth for icons
 * (unified feature spec §2.5, APP-022; ports the legacy Flutter
 * `core/theme/app_icons.dart` token architecture).
 *
 * Solar is BitOS's product icon language. This native facade preserves the
 * semantic token while using an SF Symbol where that is the better platform
 * convention (especially Share). A reviewed bundled Solar vector may replace
 * an individual backing without changing any feature call site. See
 * docs/native/icon-system.md and THIRD_PARTY_NOTICES.md.
 *
 * Naming: filled variant = active/on states; outline = inactive.
 */
enum AppIcons {
    /// Resolves a semantic token to its reviewed Solar asset when one is
    /// bundled; all remaining tokens retain their native SF Symbol backing.
    /// Keeping this decision here prevents feature views from knowing asset
    /// filenames or using raw SVGs.
    static func image(for symbol: String) -> Image {
        switch symbol {
        case heart: Image("SolarHeartLinear")
        case heartFill: Image("SolarHeartBold")
        case bookmark: Image("SolarBookmarkLinear")
        case bookmarkFill: Image("SolarBookmarkBold")
        case comment: Image("SolarChatRoundLinear")
        case repost: Image("SolarRepeatLinear")
        case zap: Image("SolarBoltLinear")
        case more: Image("SolarMenuDotsLinear")
        case textStyle: Image("SolarTextLinear")
        case sticker: Image("SolarEmojiFunnyCircleLinear")
        case looks: Image("SolarPaletteLinear")
        case sfx: Image("SolarSoundwaveLinear")
        case musicNote: Image("SolarMusicNoteLinear")
        case warningTriangle: Image("SolarDangerTriangleLinear")
        case undo: Image("SolarUndoLeftRoundLinear")
        case check: Image("SolarCheckCircleLinear")
        case checkCircle: Image("SolarCheckCircleLinear")
        case close: Image("SolarCloseCircleLinear")
        case copy: Image("SolarCopyLinear")
        case delete: Image("SolarTrashLinear")
        case send: Image("SolarPlainLinear")
        case pen: Image("SolarPenLinear")
        case refresh: Image("SolarRefreshLinear")
        case add, create: Image("SolarAddCircleLinear")
        case search: Image("SolarMagnifierLinear")
        case filter: Image("SolarFilterLinear")
        case appsGrid: Image("SolarWidgetLinear")
        case settings: Image("SolarSettingsLinear")
        case sparkles: Image("SolarMagicWandLinear")
        case people: Image("SolarUsersLinear")
        case user, userProfile: Image("SolarUserLinear")
        case qrCode: Image("SolarQrLinear")
        case shieldCheck: Image("SolarShieldCheckLinear")
        case eyeClosed: Image("SolarEyeClosedLinear")
        case chart: Image("SolarChartLinear")
        case hashtag: Image("SolarHashtagLinear")
        case emoji: Image("SolarEmojiFunnyCircleLinear")
        case gifFilm: Image("SolarFilmLinear")
        case link: Image("SolarLinkCircleLinear")
        case home: Image("SolarHomeLinear")
        case bitz, playCircle: Image("SolarPlayCircleLinear")
        case inbox: Image("SolarBellLinear")
        case chat: Image("SolarChatRoundLinear")
        case globe: Image("SolarGlobalLinear")
        case logout: Image("SolarLogoutLinear")
        case camera: Image("SolarCameraLinear")
        case photo: Image("SolarGalleryLinear")
        case video: Image("SolarVideoCameraLinear")
        case play: Image("SolarPlayLinear")
        case pause: Image("SolarPauseLinear")
        // Camera capture chrome (spec §3.19 record screen).
        case torch: Image("SolarFlashlightOnLinear")
        case timer: Image("SolarStopwatchLinear")
        case clockCircle: Image("SolarClockCircleLinear")
        case cameraRotate: Image("SolarCameraRotateLinear")
        // Settings-only semantic tokens.
        case "hand.raised.fill", "shield.lefthalf.filled": Image("SolarShieldLinear")
        case "paintpalette.fill": Image("SolarPaletteLinear")
        case "bolt.horizontal.fill": Image("SolarRocketLinear")
        case "photo.fill": Image("SolarGalleryLinear")
        case "antenna.radiowaves.left.and.right": Image("SolarTransmissionLinear")
        case "questionmark.circle.fill": Image("SolarQuestionCircleLinear")
        case "doc.text.fill": Image("SolarInfoCircleLinear")
        default: Image(systemName: symbol)
        }
    }

    // ── Feed actions (Flutter: heart/bookmark/repost/comment/share) ──
    static let heart = "heart"
    static let heartFill = "heart.fill"
    static let bookmark = "bookmark"
    static let bookmarkFill = "bookmark.fill"
    static let repost = "arrow.2.squarepath"          // Solar: refreshSquare? web: lucide repeat
    static let comment = "bubble.right"               // Solar: chatSquare
    static let share = "square.and.arrow.up"
    static let zap = "bolt.fill"                      // Solar: bolt; zap amber

    // ── Actions (Flutter AppIcons parity) ────────────────────────────
    static let more = "ellipsis"                      // Solar: menuDots
    static let check = "checkmark"
    static let checkCircle = "checkmark.circle.fill"  // Solar: checkCircle
    static let close = "xmark"                        // Solar: closeCircle
    static let copy = "doc.on.doc"                    // Solar: copy
    static let delete = "trash"                       // Solar: trashBinMinimalistic
    static let send = "paperplane.fill"               // Solar: plain
    static let pen = "pencil"                         // compose / new note
    static let refresh = "arrow.clockwise"            // Solar: refresh
    static let undo = "arrow.uturn.backward"          // Solar: undo (editor history)
    static let save = "square.and.arrow.down"         // Solar: save (editor export to Photos)
    // ── Studio / Create hub (spec §3.17/§3.19) ───────────────────────
    static let musicNote = "musicNote"             // Solar: music (sound library)
    static let remix = "sparkles"                  // Solar: magicWand (studio remix)
    static let textStyle = "textStyle"             // Solar: text (meme caption tool)
    static let sticker = "sticker"                 // Solar: emojiFunnyCircle (sticker tool)
    static let looks = "looks"                     // Solar: palette (color grade tool)
    static let sfx = "sfx"                         // Solar: soundwave (SFX tool)
    static let warningTriangle = "warningTriangle" // Solar: dangerTriangle (variant warnings)
    static let add = "plus"                           // Solar: addCircle
    static let search = "magnifyingglass"
    static let filter = "line.3.horizontal.decrease"  // feed content filter
    static let appsGrid = "square.grid.2x2"           // More hub entry
    static let arrowUp = "arrow.up"                   // new-notes reveal (X parity; Solar swap-in later)
    static let settings = "gearshape"

    // Home feed tabs (legacy Flutter parity: sparkles / users).
    static let sparkles = "sparkles"                   // For you
    static let people = "person.2"                     // Following

    // ── Moderation / safety (spec §3.5 card menus) ───────────────────
    static let mute = "speaker.slash"
    static let reportSpam = "exclamationmark.bubble"
    static let reportIllicit = "exclamationmark.octagon"
    static let reportHarassment = "hand.raised"
    static let block = "hand.raised.slash"            // Solar: closest: closeCircle

    // ── Composer toolbar (web Composer parity: shield-check PoW,
    // eye-off sensitive, bar-chart poll, film GIF, hash, smile) ──────
    static let shieldCheck = "shieldCheck"
    static let eyeClosed = "eyeClosed"
    static let eye = "eye"                             // own-story view count
    static let chart = "chart"
    static let hashtag = "hashtag"
    static let emoji = "emoji"
    static let gifFilm = "gifFilm"
    static let link = "link"

    // ── Identity / navigation ────────────────────────────────────────
    static let user = "person.fill"                   // Solar: user
    static let userProfile = "person.crop.circle"
    static let qrCode = "qrcode"                      // Solar: qrCode
    static let home = "house.fill"                  // Home tab (notes feed)
    static let bitz = "play.rectangle.fill"           // Bitz tab (reels)
    static let discover = "magnifyingglass"
    static let create = "plus.square.fill"
    static let inbox = "bell.fill"
    static let chat = "message.fill"                  // DMs
    static let lock = "lock.fill"                     // NIP-17 encryption indicator
    static let chevronLeft = "chevron.left"            // chat / subpage back
    static let chevronRight = "chevron.right"          // disclosure rows
    static let globe = "globe"                        // Solar: global
    static let logout = "rectangle.portrait.and.arrow.right"

    // ── Media ────────────────────────────────────────────────────────
    static let camera = "camera.fill"
    static let photo = "photo.on.rectangle"
    static let video = "video.fill"
    static let play = "play.fill"
    static let playCircle = "play.circle.fill"   // Solar: playCircle (video badge)
    static let pause = "pause.fill"
    static let brokenImage = "photo.badge.exclamationmark"  // failed media tile
    static let info = "info.circle"                          // picker footer
    static let branch = "arrow.triangle.branch"              // orphan reply indicator

    // ── Bitz player controls (spec §3.7) ─────────────────────────────
    static let soundOn = "speaker.wave.2.fill"        // unmuted state
    static let back10 = "gobackward.10"               // −10 s seek pill
    static let forward10 = "goforward.10"             // +10 s seek pill

    // ── Camera capture chrome (spec §3.19 record screen) ─────────────
    static let torch = "torch"               // Solar: flashlightOn (torch toggle)
    static let timer = "timer"               // Solar: stopwatch (self-timer chip)
    static let clockCircle = "clockCircle"   // Solar: clockCircle (duration cap)
    static let cameraRotate = "cameraRotate" // Solar: cameraRotate (flip lens)
    // Solar Linear has no 3×3 grid glyph; native square.grid.3x3 by the
    // icon-system.md platform-convention rule.
    static let gridGuides = "square.grid.3x3"

    // ── Editor shell (MSU-014): one glyph per tool, no collisions ─────
    static let captions = "captions.bubble"           // classic top/bottom captions
    static let timeline = "timeline.selection"        // timeline workspace entry
    static let frameTray = "rectangle.stack"          // GIF frame tray
    static let gifLibrary = "photo.stack"             // GIF library browse
    static let durationTimer = "timer"                // duration / frame hold
    static let keyboardKeys = "keyboard"              // shortcuts reference
    static let split = "scissors.badge.ellipsis"      // split clip at playhead
    static let schedule = "clock.arrow.circlepath"    // overlay timing window
    static let moreIcon = "ellipsis"                  // More overflow

    /// Catalogue icon key → the platform glyph (MSU-014 parity table).
    /// Keys come from `MemeTools.ToolId.iconKey` via the bridge.
    static func symbol(forCatalogIcon key: String) -> String {
        switch key {
        case "media": return photo
        case "text": return textStyle
        case "sticker": return sticker
        case "sound": return sfx
        case "look": return looks
        case "captions": return captions
        case "canvas": return "rectangle.on.rectangle"
        case "layers": return "square.3.layers.3d"
        case "draw": return pen
        case "trim": return "scissors"
        case "speed": return durationTimer
        case "clips": return video
        case "timeline": return timeline
        case "volume": return "speaker.wave.2"
        case "sfx": return sfx
        case "frames": return frameTray
        case "gifs": return gifLibrary
        case "duration": return durationTimer
        case "batch": return remix
        case "shortcuts": return keyboardKeys
        case "edit": return pen
        case "time": return schedule
        case "duplicate": return "square.on.square"
        case "forward": return "square.2.layers.3d.top.filled"
        case "backward": return "square.2.layers.3d.bottom.filled"
        case "split": return split
        case "mute": return "speaker.slash"
        case "hold": return durationTimer
        case "move-left": return "chevron.left"
        case "move-right": return "chevron.right"
        case "delete": return "trash"
        case "more": return moreIcon
        default: return "circle"
        }
    }
}

File: apps/android/src/main/kotlin/space/bitos/app/ui/theme/AppIcons.kt
package space.bitos.app.ui.theme

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.ChatBubble
import androidx.compose.material.icons.rounded.ChevronLeft
import androidx.compose.material.icons.rounded.Explore
import androidx.compose.material.icons.rounded.KeyboardArrowDown
import androidx.compose.material.icons.rounded.KeyboardArrowUp
import androidx.compose.material.icons.rounded.Visibility
import androidx.compose.material.icons.rounded.VisibilityOff
import androidx.compose.material.icons.rounded.ZoomIn
import androidx.compose.material.icons.rounded.ZoomOut
import androidx.compose.material.icons.automirrored.rounded.ArrowBack
import androidx.compose.material.icons.automirrored.rounded.Send
import androidx.compose.material.icons.automirrored.rounded.Undo
import androidx.compose.material.icons.rounded.Add
import androidx.compose.material.icons.rounded.ArrowUpward
import androidx.compose.material.icons.rounded.Download
import androidx.compose.material.icons.rounded.AutoAwesome
import androidx.compose.material.icons.rounded.AutoFixHigh
import androidx.compose.material.icons.rounded.Bolt
import androidx.compose.material.icons.rounded.Bookmark
import androidx.compose.material.icons.rounded.BookmarkBorder
import androidx.compose.material.icons.rounded.CameraAlt
import androidx.compose.material.icons.rounded.ChatBubbleOutline
import androidx.compose.material.icons.rounded.Check
import androidx.compose.material.icons.rounded.CheckCircle
import androidx.compose.material.icons.rounded.Close
import androidx.compose.material.icons.rounded.ContentCopy
import androidx.compose.material.icons.rounded.Delete
import androidx.compose.material.icons.rounded.Edit
import androidx.compose.material.icons.rounded.ExitToApp
import androidx.compose.material.icons.rounded.Favorite
import androidx.compose.material.icons.rounded.FavoriteBorder
import androidx.compose.material.icons.rounded.GridView
import androidx.compose.material.icons.rounded.Gif
import androidx.compose.material.icons.rounded.Lock
import androidx.compose.material.icons.rounded.AspectRatio
import androidx.compose.material.icons.rounded.Crop
import androidx.compose.material.icons.rounded.Layers
import androidx.compose.material.icons.rounded.Loop
import androidx.compose.material.icons.rounded.Movie
import androidx.compose.material.icons.rounded.Palette
import androidx.compose.material.icons.rounded.TextFields
import androidx.compose.material.icons.rounded.Wallpaper
import androidx.compose.material.icons.rounded.BrokenImage
import androidx.compose.material.icons.rounded.PanTool
import androidx.compose.material.icons.rounded.Home
import androidx.compose.material.icons.rounded.HowToVote
import androidx.compose.material.icons.rounded.MoreHoriz
import androidx.compose.material.icons.rounded.MusicNote
import androidx.compose.material.icons.rounded.LocalFireDepartment
import androidx.compose.material.icons.rounded.Numbers
import androidx.compose.material.icons.rounded.ChevronRight
import androidx.compose.material.icons.rounded.Notifications
import androidx.compose.material.icons.rounded.People
import androidx.compose.material.icons.rounded.Person
import androidx.compose.material.icons.rounded.Link
import androidx.compose.material.icons.rounded.PlayArrow
import androidx.compose.material.icons.rounded.PlayCircle
import androidx.compose.material.icons.rounded.QrCode2
import androidx.compose.material.icons.rounded.Repeat
import androidx.compose.material.icons.rounded.Search
import androidx.compose.material.icons.rounded.Save
import androidx.compose.material.icons.rounded.Share
import androidx.compose.material.icons.rounded.Tune
import androidx.compose.material.icons.rounded.ContentCut
import androidx.compose.material.icons.rounded.Speed
import androidx.compose.material.icons.automirrored.rounded.Redo
import androidx.compose.material.icons.rounded.VolumeOff
import androidx.compose.material.icons.rounded.Warning
import androidx.compose.material.icons.rounded.ErrorOutline
import androidx.compose.material.icons.rounded.Public
import androidx.compose.material.icons.rounded.Videocam
import androidx.compose.material.icons.rounded.PhotoLibrary
import androidx.compose.material.icons.rounded.Refresh
import androidx.compose.material.icons.rounded.Pause
import androidx.compose.material.icons.automirrored.rounded.VolumeUp
import androidx.compose.material.icons.rounded.Replay10
import androidx.compose.material.icons.rounded.Forward10
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * AppIcons — the application's single source of truth for icons
 * (unified feature spec §2.5, APP-022; ports the legacy Flutter
 * `core/theme/app_icons.dart` token architecture).
 *
 * Solar is BitOS's product icon language. This native facade preserves the
 * semantic token while using Material where that is the better platform
 * convention (especially Android sharing). A reviewed bundled Solar vector
 * may replace an individual backing without changing any feature call site.
 * See docs/native/icon-system.md and THIRD_PARTY_NOTICES.md.
 */
object AppIcons {

    // ── Feed actions ─────────────────────────────────────────────────
    val Heart = Icons.Rounded.Favorite
    val HeartOutline = Icons.Rounded.FavoriteBorder
    val Bookmark = Icons.Rounded.Bookmark
    val BookmarkOutline = Icons.Rounded.BookmarkBorder
    val Repost = Icons.Rounded.Repeat           // Solar: refreshSquare? web: lucide repeat
    val Comment = Icons.Rounded.ChatBubbleOutline // Solar: chatSquare
    val Share = Icons.Rounded.Share
    val Zap = Icons.Rounded.Bolt                // zap amber

    // ── Actions (Flutter AppIcons parity) ────────────────────────────
    val More = Icons.Rounded.MoreHoriz          // Solar: menuDots
    val Check = Icons.Rounded.Check
    val CheckCircle = Icons.Rounded.CheckCircle // Solar: checkCircle (verified badge)
    val Close = Icons.Rounded.Close             // Solar: closeCircle
    val Copy = Icons.Rounded.ContentCopy        // Solar: copy
    val Delete = Icons.Rounded.Delete           // Solar: trashBinMinimalistic
    val Send = Icons.AutoMirrored.Rounded.Send  // Solar: plain
    val Back = Icons.AutoMirrored.Rounded.ArrowBack // chat / subpage back
    val Lock = Icons.Rounded.Lock              // NIP-17 encryption indicator
    val Pen = Icons.Rounded.Edit                 // compose / new note (Solar: pen)
    val Refresh = Icons.Rounded.Refresh         // Solar: refresh
    val ArrowUp = Icons.Rounded.ArrowUpward      // new-notes reveal (X parity)
    val Add = Icons.Rounded.Add                 // Solar: addCircle
    val Search = Icons.Rounded.Search
    val Filter = Icons.Rounded.Tune             // feed content filter
    val AppsGrid = Icons.Rounded.GridView       // More hub entry

    // ── Moderation / safety (spec §3.5 card menus) ───────────────────
    val Mute = Icons.Rounded.VolumeOff
    val Poll = Icons.Rounded.HowToVote             // poll composer (legacy parity)
    val ReportSpam = Icons.Rounded.Warning
    val ReportIllicit = Icons.Rounded.ErrorOutline
    val ReportHarassment = Icons.Rounded.PanTool
    val Block = Icons.Rounded.Close             // Solar: closest: closeCircle

    // ── Bitz player controls (spec §3.7) ─────────────────────────────
    val SoundOn = Icons.AutoMirrored.Rounded.VolumeUp   // unmuted state
    val Pause = Icons.Rounded.Pause
    val Back10 = Icons.Rounded.Replay10    // −10 s seek pill
    val Forward10 = Icons.Rounded.Forward10 // +10 s seek pill

    // ── Identity / navigation ────────────────────────────────────────
    val User = Icons.Rounded.Person             // Solar: user
    val QrCode = Icons.Rounded.QrCode2          // Solar: qrCode
    val Home = Icons.Rounded.Home
    val Discover = Icons.Rounded.Search
    val Create = Icons.Rounded.Add

    // Home feed tabs (legacy Flutter parity: sparkles / users).
    val Sparkles = Icons.Rounded.AutoAwesome
    val People = Icons.Rounded.People
    val Inbox = Icons.Rounded.Notifications
    val Chat = Icons.Rounded.ChatBubble
    val Globe = Icons.Rounded.Public            // Solar: global
    val Logout = Icons.Rounded.ExitToApp

    // ── Discover (prototype `#/discover` parity) ────────────────────
    val Flame = Icons.Rounded.LocalFireDepartment // trending-hot hashtag
    val Hash = Icons.Rounded.Numbers                // hashtag glyph
    val ChevronRight = Icons.Rounded.ChevronRight // list-row affordance
    val ChevronLeft = Icons.Rounded.ChevronLeft   // list-row affordance (mirrored)

    // ── Overlay manipulation (MUX-03 selection controls) ────────────
    val NudgeLeft = Icons.Rounded.ChevronLeft       // Solar: altArrowLeft
    val NudgeRight = Icons.Rounded.ChevronRight     // Solar: altArrowRight
    val NudgeUp = Icons.Rounded.KeyboardArrowUp     // Solar: altArrowUp
    val NudgeDown = Icons.Rounded.KeyboardArrowDown // Solar: altArrowDown
    val ZoomOut = Icons.Rounded.ZoomOut             // Solar: magniferZoomOut
    val ZoomIn = Icons.Rounded.ZoomIn               // Solar: magniferZoomIn

    // ── Studio / Create hub (spec §3.17/§3.19) ───────────────────────
    val MusicNote = Icons.Rounded.MusicNote     // Solar: music (sound library)
    val Remix = Icons.Rounded.AutoFixHigh       // Solar: magicWand (studio remix)
    val Undo = Icons.AutoMirrored.Rounded.Undo  // Solar: undo (editor history)
    val Redo = Icons.AutoMirrored.Rounded.Redo  // Solar: undoRight (suite dock)
    val Download = Icons.Rounded.Download       // Solar: save (editor export)
    val SaveDraft = Icons.Rounded.Save          // editor project draft (distinct from export)
    val Scissors = Icons.Rounded.ContentCut     // V2 suite: Trim tool chip
    val Speed = Icons.Rounded.Speed             // V2 suite: Speed tool chip
    val Film = Icons.Rounded.Movie              // prototype editor: Split clip tool
    val Layer = Icons.Rounded.Layers            // prototype editor: Layer clip tool
    val Loop = Icons.Rounded.Loop               // prototype editor: GIF loop bar slot
    val Crop = Icons.Rounded.Crop               // prototype editor: Crop bar slot (wave 2)
    val Ratio = Icons.Rounded.AspectRatio       // prototype editor: Canvas/Ratio bar slot
    val Looks = Icons.Rounded.Palette           // prototype editor: Filter/Effects bar slot
    val TextGlyph = Icons.Rounded.TextFields    // prototype editor: Text bar slot
    val Wallpaper = Icons.Rounded.Wallpaper     // prototype editor: Background bar slot

    // ── Media ────────────────────────────────────────────────────────
    val Camera = Icons.Rounded.CameraAlt
    val Photo = Icons.Rounded.PhotoLibrary
    val Video = Icons.Rounded.Videocam
    val Play = Icons.Rounded.PlayArrow
    val PlayCircle = Icons.Rounded.PlayCircle   // Solar: playCircle (video badge)
    val Gif = Icons.Rounded.Gif                       // GIF picker (legacy parity)
    val BrokenImage = Icons.Rounded.BrokenImage       // failed media tile
    val Link = Icons.Rounded.Link                     // Solar: linkCircle (profile link)
    val Compass = Icons.Rounded.Explore          // Solar: compass (public stories lane)
    val Visibility = Icons.Rounded.Visibility        // own-story view count
    val VisibilityOff = Icons.Rounded.VisibilityOff  // Solar: eyeClosed (sensitive gate)

    /** Convenience for components taking optional leading icons. */
    val none: ImageVector? = null
}
